ttnn/cpp/ttnn/operations/pool/maxpool/max_pool2d.hpp

```diff
@@ -1,37 +0,0 @@
-// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
```

Comment:
I think it would be better to rename this file to `generic_pools.hpp` instead of deleting this and creating new file -- so that we can carry on the git history.

Reply:
I renamed the file when creating this PR. But it seems that git isn't able to detect renaming when it also includes many modifications.

I decided to create a new PR https://github.com/tenstorrent/tt-metal/pull/15561 to refactor `max_pool2d` into generic pools first, which git can track the renaming and also makes smaller PRs

Reply:
Done. The prerequisite PR https://github.com/tenstorrent/tt-metal/pull/15561 has been merged and there is no separated delete and create files in git history